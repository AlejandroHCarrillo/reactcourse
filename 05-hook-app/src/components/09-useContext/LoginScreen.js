import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const { user, setUser } = useContext(UserContext);

    const handleLogin = () =>{
        setUser({
                    id: 1234,
        nombre: 'Alejandro',
        email: 'alex@hotmail.com'
        });
    };

    return (
        <>
            <h1>Login</h1>
            <hr/>
            <button className="btn btn-success" 
                onClick={ handleLogin }
            >
                Login
            </button>
        </>
    )
}
