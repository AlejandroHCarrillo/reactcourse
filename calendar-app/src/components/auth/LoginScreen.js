import React from 'react'

export const LoginScreen = () => {
    return (
        <>          
            <h1>LoginScreen</h1>
            <h2>otra cosa</h2>
        </>
    )
}
