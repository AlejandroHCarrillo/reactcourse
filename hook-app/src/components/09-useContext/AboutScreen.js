import React, { useContext } from 'react'
import "../css/style.css"
import { UserContext } from './UserContext';

export const AboutScreen = () => {
    const { user, setUser } = useContext(UserContext);

    const handleLogout = () =>{
        setUser({});
    };

    return (
        <>
            <h1>About</h1>
            <hr/> 
            <pre>
                { JSON.stringify(user, null, 2) }
            </pre>

            <button className="btn btn-warning" 
                onClick={ handleLogout }
            >
                Logiout
            </button>
        </>
    )
}
