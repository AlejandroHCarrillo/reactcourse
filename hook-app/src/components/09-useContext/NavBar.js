import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <Link className="navbar-brand" to='/'>Navbar</Link>

                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink to='/'      activeClassName="active" className="nav-link" exact >Home</NavLink>
                    <NavLink to="/about" activeClassName="active" className="nav-link" exact >About</NavLink>
                    <NavLink to="/login" activeClassName="active" className="nav-link" exact >Login</NavLink>
                </div>
                </div>
            </div>
        </nav>

    )
}
