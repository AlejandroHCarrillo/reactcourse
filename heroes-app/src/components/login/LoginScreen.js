import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginScreen = ({ history }) => {

    const { dispatch } =  useContext(AuthContext);
    
    const handleClick = () => {
        // console.log("Login click");
        dispatch({
            type: types.login,
            payload: {
                name: "Alejandro",
                logged: true
            }
        });
        
        const lastPath = localStorage.getItem('lastPath')||'/home';
        if(lastPath!==''){
            history.replace(lastPath);
        } else{
            history.replace('/home');
        }
    };

    return (
        <div>
            <h1>Login</h1>
            <hr/>

            <button className="btn btn-primary"
                onClick={ handleClick }
            >
                Login
            </button>
        </div>
    )
}
