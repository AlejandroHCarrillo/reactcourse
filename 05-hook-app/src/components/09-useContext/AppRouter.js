import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link,
    Redirect
  } from "react-router-dom";

import {AboutScreen} from "./AboutScreen"
import {LoginScreen} from "./LoginScreen"
import {HomeScreen} from "./HomeScreen"
import { NavBar } from './NavBar';

export const AppRouter = () => {
    
    return (
        <Router>
            <div className="container">
                <NavBar/>
                <Switch>
                    <Route exact path="/" component={HomeScreen} />
                    
                    <Route exact path="/about" component={AboutScreen} />
                    <Route exact path="/login" component={LoginScreen} />

                    {/* Cuando una ruta no existe te direcciona a Home */}
                    {/* <Route component={ HomeScreen } /> */}
                    {/* Segundo metodo para redireccionar al home */}
                    <Redirect to="/"/>
                </Switch>
            </div>
        </Router>

    )
}
