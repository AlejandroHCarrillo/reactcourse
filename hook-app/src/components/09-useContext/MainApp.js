import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainScreen = () => {
    // const user = {
    //     id: 1234,
    //     nombre: 'Alejandro',
    //     email: 'alex@hotmail.com'
    // }
    const [user, setUser] = useState({});

    return (
        <UserContext.Provider value={   {
                                            user, 
                                            setUser 
                                        }
                                    }
        >
            <AppRouter/>
        </UserContext.Provider>
    )
}
