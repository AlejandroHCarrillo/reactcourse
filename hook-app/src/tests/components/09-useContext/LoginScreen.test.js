import { mount, shallow } from 'enzyme';
import React from 'react';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

// "mount" no sirve con react 17
// Mount sirve para rederizar los componentes hijos
// como no funciona estas pruebas van a fallar
describe('Pruebas en LoginScreen', () => {
    const user = {
        id: 1234,
        nombre: 'Alejandro',
        email: 'alex@hotmail.com'
    };

    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{ setUser }}>
            <LoginScreen />
        </UserContext.Provider>
    );
    // const wrapper = shallow(<LoginScreen />);

    test('should de mostrar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should de ejecutar el setUser con el argumento esperado', () => {
        wrapper.find('button').prop('onClick')();

        expect(setUser).toHaveBeenCalledWith({
            id: 1234,
            nombre: 'Alejandro',
            email: 'alex@hotmail.com'
        });
    })
    
    
})
