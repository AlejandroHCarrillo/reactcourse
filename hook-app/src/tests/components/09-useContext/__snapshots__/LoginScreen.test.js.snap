// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen should de mostrar el componente 1`] = `<LoginScreen />`;
